lxd/device/device_utils_network.go

```diff
@@ -901,18 +901,36 @@ func networkSRIOVRestoreVF(d deviceCommon, useSpoofCheck bool, volatile map[stri
 func networkPCIBindWaitInterface(pciDev pcidev.Device, ifName string) error {
 	var err error
 
+	waitDuration := time.Second * 10
+	waitUntil := time.Now().Add(waitDuration)
+
 	// Keep requesting the device driver be probed in case it was not ready previously or the expected
 	// interface has not appeared yet. The device can be probed multiple times safely.
-	for i := 0; i < 10; i++ {
+
+	i := 0
+	for {
 		err = pcidev.DeviceProbe(pciDev)
 		if err == nil && network.InterfaceExists(ifName) {
 			return nil
 		}
 
-		time.Sleep(50 * time.Millisecond)
-	}
+		if time.Now().After(waitUntil) {
+			if err != nil {
+				return fmt.Errorf("Failed bind interface %q after %v: %w", ifName, waitDuration, err)
```

Comment:
```suggestion
				return fmt.Errorf("Failed to bind interface %q after %v: %w", ifName, waitDuration, err)
```

Just a small suggestion, this caught my eye.

Reply:
yes good point ill change to "Failed binding interface..."